### PR TITLE
4 get/api/articles/article

### PR DESCRIPTION
**Pull request by holyseo**
Sorry about messing around. This must be a good experience that I can get familiar with git version control. Thanks!

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -54,18 +55,44 @@ describe("GET/api/articles/:article_id", () => {
...
-          article_img_url:
-            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
-        };
-        expect(sample).toEqual(body[0]);
+        const { articles } = body;
+        articles.forEach((article) => {
+          expect(article).toMatchObject({
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: expect.any(Number),
```
Comment (popatre) [inline __tests__/app.test.js]: You need to expect the article id that you've requested in the url

File: __tests__/app.test.js [hunk 2]
```diff
@@ -54,18 +55,44 @@ describe("GET/api/articles/:article_id", () => {
...
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+          });
+        });
+        console.log(articles);
```
Comment (popatre) [inline __tests__/app.test.js:94]: remove console log

File: controllers/getArticlesByIdController.js [hunk 3]
```diff
@@ -3,7 +3,6 @@ const { selectArticleById } = require("../models/articles");
 exports.getArticlesById = (req, res) => {
   const { article_id } = req.params;
   selectArticleById(article_id).then((article) => {
-    console.log(article);
-    res.status(200).send(article);
+    res.status(200).send({ articles: article });
```
Comment (popatre) [inline controllers/getArticlesByIdController.js]: you only get one article, so should be a key of article.

File: models/articles.js [hunk 4]
```diff
@@ -7,3 +7,9 @@ exports.selectArticleById = (id) => {
       return rows;
     });
 };
+
+exports.selectAllArticles = () => {
+  return db.query(`SELECT * FROM ARTICLES ;`).then(({ rows }) => {
```
Comment (popatre) [inline models/articles.js]: need to make sure the article has a comment count, and is in the order asked for in the ticket.

**Review by popatre — CHANGES_REQUESTED** [review]
Not sure which ticket this is for as your pr says 4, but seems to have 5 as well. 
Please see my comments. 

ticket 4 - need to make sure you're getting the correct article id. 

For ticket 5, we need a comment_count

In addition:

the articles should be sorted by date in descending order.
there should not be a body property present on any of the article objects.

You'll need to make sure you've tested they're in the correct order

**Comment by holyseo** [comment]
Hi Jonathan,

I messed git version controls between task 4 and 5, but you kindly reviewed my code for both tickets 4 and 5. I have pushed them again with correction based on your suggestion. Many thanks!

**Review by ghost — APPROVED** [review]
Looks like you're missing the update on your endpoints.json, could you make those changes then go ahead and merge.

**Pull request closed without merging**